File: sort_algorithms/selection/selectionSort.js
/**
 * This function sort an array of elements using the selection sort algorithm
 * 
 * @param {*} array 
 * @returns The array sorted
 */

function selectionSort(array) {
    let array_length = array.length;

    for (let i = 0; i < array_length; i++) {
        let min_element_index = i;
        
        // Looks for the index of the minimum element in the array
        for (let j = i + 1; j < array_length; j++) {
            if (array[j] < array[min_element_index]) {
                min_element_index = j;
            }
        }

        // Swap positions of the minimum element with the current element
        let temp = array[i];
        array[i] = array[min_element_index];
        array[min_element_index] = temp;
    }

    return array;
}

numbers = [99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0];
console.log("Original numbers: [" + numbers.join(", ") + "]");

sortedNumbers = selectionSort(numbers);
console.log("Sorted numbers:   [" + sortedNumbers.join(", ") + "]\n");


words = ["Manzana", "Naranja", "Hola", "Mundo", "Emiliano", "UPIICSA", "Aura", "Cielo", "Oscar"]
console.log("Original words: [" + words.join(", ") + "]");

sortedWords = selectionSort(words);
console.log("Sorted words:   [" + sortedWords.join(", ") + "]");

File: sort_algorithms/selection/selection_sort.py
def selection_sort(array):
    """
    This function implements the selection sort algorithm to sort a list of integers or words.

    Args:
        array (list[]): The list of elements to sort.
    Return:
        array (list[]): The sorted list of elements.
    """

    array_length = len(array)

    for i in range(array_length):
        # Looks for the minimum element
        min_element_index = i

        for j in range(i+1, array_length):
            if array[j] < array[min_element_index]:
                min_element_index = j

        # Swap positions using multiple assignment
        array[i], array[min_element_index] = array[min_element_index], array[i]
    return array


numbers = [23, 5, 9, 10, 1, 3, 7]
words = ["Hola", "Mundo", "Adios", "Como", "Estas", "Aqui"]

print(selection_sort(numbers))
print(selection_sort(words))
